gpflow/base.py

```diff
@@ -307,6 +304,9 @@ def _run_op(a, *args, **kwargs):
 Parameter._OverloadAllOperators()
 tf.register_tensor_conversion_function(Parameter, lambda x, *args, **kwds: x.read_value())
 
+TensorLike = Union[np.ndarray, tf.Tensor, tf.Variable, Parameter]
+TensorLikeTypes = (tf.Tensor, tf.Variable, np.ndarray, Parameter)
```

Comment:
Would it not make more sense to have the types together at the top of the file (as it was before)?
Alternatively, it might be helpful to collate them in a `gpflow.types` module (but perhaps do that in a separate PR?)... :thinking:

Reply:
It might also be helpful to add a comment explaining why we've got two things here that are almost exactly the same but not quite

Reply:
at the top of the file would be nice, but they need to be after the `Parameter` class to work

gpflow.types module? perhaps

comment? yes. will do